### PR TITLE
Ck ce estimated next purchase

### LINKED ISSUE #10: 10. As an item, I want my estimated next purchase date to be computed at the time my purchase is recorded in the database so the app can learn how often I buy different items.
AC:
      * When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database
      The script `calculateEstimate` from `@the-collab-lab/shopping-list-utils` should be used to calculate the next estimated purchase interval. This should already be installed in your project from issue #3.
      Notes:
      * Think through the data that is needed as inputs to the function in `estimates.js` and how to use the output.
      * It is likely at this point you'll need to adjust the schema for an item to accommodate the data points required.

### PR DESCRIPTION
**Pull request by cynthiaeddy**
## Description

This PR incorporates the Calculate Estimate function to calculate number of days between the last and next purchase.

New data points have been added when an item is created in able for this function to work. 
The data points are as follows: 
  'created item' which records in the db the time the item is created.
  'estimated purchase interval' which is initially set to 0.
  'total purchases' which is initially set to 0.
  
Also added the parameter item to the handleChecked function in order to calculate 'daysSinceLastPurchased' variable.

## Related Issue

Closes #10

AC:
* When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database
The script calculateEstimate from @the-collab-lab/shopping-list-utils should be used to calculate the next estimated purchase interval. This should already be installed in your project from issue https://github.com/the-collab-lab/tcl-38-smart-shopping-list/issues/3.
Notes:
* Think through the data that is needed as inputs to the function in estimates.js and how to use the output.
* It is likely at this point you'll need to adjust the schema for an item to accommodate the data points required.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

without the function:
Screen Shot 2022-02-09 at 5 03 34 PM

with the function:
Screen Shot 2022-02-09 at 5 04 23 PM

## Testing Steps / QA Criteria

Pull down the branch, add an item, then purchase it.
Go to db and manipulate the purchase date. 
Purchase the item again, the check the db for the updated estimated purchase interval.

### CODE REVIEW

**Review by diyanamendoza — APPROVED** [review]

**Comment by diyanamendoza** [comment]
looks and works great!!

**Review by k-monnik — APPROVED** [review]
This looks like it was a challenge with all the math, but y'all did a great job! Works well!

File: src/App.js [hunk 1]
```diff
@@ -10,13 +10,6 @@ import AddItem from './components/AddItem';
...
 import Home from './pages/Home';
 
-import {
-  getToken,
-  words,
-  calculateEstimate,
-} from '@the-collab-lab/shopping-list-utils';
-import { useNavigate } from 'react-router-dom';
-
 function App() {
```
Comment (aprilleone) [inline src/App.js:12]: with this removed here, you probably could have resolved the issue #29 along with this feature. I think that would be fine.

File: src/components/ItemList.js [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
+      ),
       'last purchased': serverTimestamp(),
+      'total purchases': (item.data['total purchases'] += 1),
     });
   };
 
   const within24Hours = (item) => {
     // Seconds in 24 hours
-    // 60 x 60 x 24 = 86400
+    // 60 x 60 x 24 = 86400å
```
Comment (aprilleone) [inline src/components/ItemList.js]: is the character at the end of the line an unintended artifact?

File: src/components/ItemList.js [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
     const now = Timestamp.now();
     if (item.data['last purchased']) {
       if (now.seconds - item.data['last purchased'].seconds < 86400) {
```
Comment (aprilleone) [inline src/components/ItemList.js]: slick formating. Question, is the const 'now' used only once? Does it need to be a const or could you apply Timestamp.now().seconds?

**Review by aprilleone — APPROVED** [review]
Excellent. Approved!

There are some small considerations, but it's great.

File: src/components/ItemList.js (hunk 2 above)
Comment (cynthiaeddy) [reply src/components/ItemList.js]: yup, removed.

File: src/components/ItemList.js (hunk 3 above)
Comment (cynthiaeddy) [reply src/components/ItemList.js]: reconfigured to remove the variable for now as suggested

File: src/App.js (hunk 1 above)
Comment (chriskorsak) [reply src/App.js:12]: We were on the fence as to combining issues, even though #29 was so straight forward. Would love to hear advice when this is an acceptable practice, or maybe it's just a case by case basis?

**Comment by skylerwshaw** [comment]
Spamming the PR with an example of Github-flavor markdown tables:
| Thing A | Thing B |
| --- | --- |
| peimei_ninja | magic |

**Pull request merged**